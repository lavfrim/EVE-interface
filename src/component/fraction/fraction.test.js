import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Fraction from './index';

const info = {
    name: 'name',
    description: 'description',
    faction_id: 500001,
    corporation_id:1000035,
    solar_system_id: 30000145,
};

describe('Fraction should render correctly', () => {
    it('with moch info, isOpen=false', () => {
        const wrapper = mount(
            <Fraction info={info} />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('with moch info, isOpen=true', () => {
        const wrapper = mount(
            <Fraction info={info} />
        );
        wrapper.setState({ isOpen: true });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    
});

describe('Fraction should have correctly reaction on click', () => {
    it('not clicked', () => {
        const wrapper = mount(
            <Fraction info={info} />
        );
        expect(wrapper.state().isOpen).toBe(false);
    });

    it('clicked', () => {
        const wrapper = mount(
            <Fraction info={info} />
        );
        wrapper.find('.fraction-card').simulate('click');
        expect(wrapper.state().isOpen).toBe(true);
    });
});

