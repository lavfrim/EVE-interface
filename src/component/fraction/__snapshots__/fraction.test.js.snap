// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fraction should render correctly with moch info, isOpen=false 1`] = `
<Fraction
  info={
    Object {
      "corporation_id": 1000035,
      "description": "description",
      "faction_id": 500001,
      "name": "name",
      "solar_system_id": 30000145,
    }
  }
>
  <div
    className="fraction-card"
    id={500001}
    onClick={[Function]}
  >
    <p
      className="fraction-card__name"
    >
      name
    </p>
  </div>
</Fraction>
`;

exports[`Fraction should render correctly with moch info, isOpen=true 1`] = `
<Fraction
  info={
    Object {
      "corporation_id": 1000035,
      "description": "description",
      "faction_id": 500001,
      "name": "name",
      "solar_system_id": 30000145,
    }
  }
>
  <div
    className="fraction-card"
    id={500001}
    onClick={[Function]}
  >
    <p
      className="fraction-card__name"
    >
      name
    </p>
    <div
      className="fraction-card__corp-link"
    >
      <p
        className="fraction-card__corp-link-description"
      >
        Corporation name:
      </p>
      <Loading>
        <span>
          Loading...
        </span>
      </Loading>
    </div>
    <p
      className="fraction-card__description"
    >
      Description: description
    </p>
    <div
      className="fraction-card__solar-sys"
    >
      <p
        className="fraction-card__solar-sys-description"
      >
        System:
      </p>
      <Loading>
        <span>
          Loading...
        </span>
      </Loading>
    </div>
  </div>
</Fraction>
`;

exports[`Fraction should render correctly with moch info, isOpen=true 2`] = `
<Fraction
  info={
    Object {
      "corporation_id": 1000035,
      "description": "description",
      "faction_id": 500001,
      "name": "name",
      "solar_system_id": 30000145,
    }
  }
>
  <div
    className="fraction-card"
    id={500001}
    onClick={[Function]}
  >
    <p
      className="fraction-card__name"
    >
      name
    </p>
    <div
      className="fraction-card__corp-link"
    >
      <p
        className="fraction-card__corp-link-description"
      >
        Corporation name:
      </p>
      <Loading>
        <span>
          Loading...
        </span>
      </Loading>
    </div>
    <p
      className="fraction-card__description"
    >
      Description: description
    </p>
    <div
      className="fraction-card__solar-sys"
    >
      <p
        className="fraction-card__solar-sys-description"
      >
        System:
      </p>
      <Loading>
        <span>
          Loading...
        </span>
      </Loading>
    </div>
  </div>
</Fraction>
`;

exports[`Fraction should render correctly with moch info, isOpen=true, connection = lost 1`] = `
<Fraction
  info={
    Object {
      "corporation_id": 1000035,
      "description": "description",
      "faction_id": 500001,
      "name": "name",
      "solar_system_id": 30000145,
    }
  }
>
  <div
    className="fraction-card"
    id={500001}
    onClick={[Function]}
  >
    <p
      className="fraction-card__name"
    >
      name
    </p>
    <div
      className="fraction-card__corp-link"
    >
      <p
        className="fraction-card__corp-link-description"
      >
        Corporation name:
      </p>
      <Loading>
        <span>
          Loading...
        </span>
      </Loading>
    </div>
    <p
      className="fraction-card__description"
    >
      Description: description
    </p>
    <div
      className="fraction-card__solar-sys"
    >
      <p
        className="fraction-card__solar-sys-description"
      >
        System:
      </p>
      <Loading>
        <span>
          Loading...
        </span>
      </Loading>
    </div>
  </div>
</Fraction>
`;

exports[`Fraction should render correctly with moch info, isOpen=true, connet 1`] = `
<Fraction
  info={
    Object {
      "corporation_id": 1000035,
      "description": "description",
      "faction_id": 500001,
      "name": "name",
      "solar_system_id": 30000145,
    }
  }
>
  <div
    className="fraction-card"
    id={500001}
    onClick={[Function]}
  >
    <p
      className="fraction-card__name"
    >
      name
    </p>
    <div
      className="fraction-card__corp-link"
    >
      <p
        className="fraction-card__corp-link-description"
      >
        Corporation name:
      </p>
      <Loading>
        <span>
          Loading...
        </span>
      </Loading>
    </div>
    <p
      className="fraction-card__description"
    >
      Description: description
    </p>
    <div
      className="fraction-card__solar-sys"
    >
      <p
        className="fraction-card__solar-sys-description"
      >
        System:
      </p>
      <Loading>
        <span>
          Loading...
        </span>
      </Loading>
    </div>
  </div>
</Fraction>
`;
